openapi/data_repository_service.swagger.yaml

```diff
@@ -192,144 +152,156 @@ definitions:
           etag              # multipart uploads to blob stores
           sha256
           sha512
-  Bundle:
+  Object:
     type: object
-    required: ['id', 'size', 'created', 'checksums', 'contents']
+    required: ['id']
     properties:
       id:
         type: string
-        description: >-
-          An identifier unique to this Data Bundle.
+        description: |-
+          An identifier unique to this `Object`.
       name:
         type: string
-        description: >-
-          A string that can be used to name a Data Bundle.
-      size:
+        description: |-
+          A string that can be used to name an `Object`.
+      type:
         type: string
-        format: int64
-        description: >-
-          The cumulative size, in bytes, of all Data Objects and Bundles listed in
-          the `contents` field.
+        enum:
+        - blob
+        - bundle
+      blob:
+        $ref: '#/definitions/BlobDescription'
+        description: |-
+          If `type` is blob, then this property must be present and the `bundle` property must be absent or null
+      bundle:
+        $ref: '#/definitions/BundleDescription'
+        description: |-
+          If `type` is bundle, then this property must be present and the `blob` property must be absent or null
+  CommonDescription:
```

Comment:
Why `-Description`? This again implies something specific in the world of Swagger/OpenAPI. What's being specified here is the 'schema' for an object (in the Swagger sense of the word 'object') — `CommonObject` seems more appropriate to me.

Reply:
I see that this is used more as a `property` than a 'base' model for inheritance, etc. `CommonProperties` might make more sense.